Update to flow 0.15

danvk opened the pull request:
See [FlowType changelog][1].

I've also taken the opportunity to re-enable type checking for underscore. This lets many types flow through which didn't before & exposed a few existing errors. Type inference seems to have gotten significantly better!

[1]: https://github.com/facebook/flow/blob/master/Changelog.md#v0150

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/hammerlab/pileup.js/247)
<!-- Reviewable:end -->

danvk commented on src/main/CoverageTrack.js:
```diff
@@ -35,10 +35,10 @@ function extractSummaryStatistics(reads: Array<SamRead>, contig: string) {
 
-  binCounts = _.map(binCounts, (count, position) => ({position: Number(position), count}));
-  binCounts = _.sortBy(binCounts, bin => bin.position);
+  var posCounts = _.map(binCounts, (count, position) => ({position: Number(position), count}));
```
FYI FlowType was having trouble tracking the changing type of this single variable.

danvk commented:
I'm going to try re-running that build; I'm not sure why it would fail.

armish commented:
just a few minor things. Otherwise, LGTM :+1:

Thanks for doing this and upgrading us!

---

Reviewed 8 of 8 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

<sup>**[lib/q.js, line 265 \[r1\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/247#-JzC0KHj-L-XAZX6P-yn)** ([raw file](https://github.com/hammerlab/pileup.js/blob/5978d2ffb53a1a642353ae108ffb496eba73c5f0/lib/q.js#L265)):</sup>
is this commented out because it doesn't play nicely with flow or some other reason? Maybe a simple comment will help clarifying it. 

---

<sup>**[src/main/CoverageTrack.js, line 27 \[r1\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/247#-JzC-xuPdGWXjib21L_h)** ([raw file](https://github.com/hammerlab/pileup.js/blob/5978d2ffb53a1a642353ae108ffb496eba73c5f0/src/main/CoverageTrack.js#L27)):</sup>
any reason for the change? 

---

<sup>**[src/main/CoverageTrack.js, line 38 \[r1\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/247#-JzBxU9ncCmXoLrafbno-r1-38)** ([raw file](https://github.com/hammerlab/pileup.js/blob/5978d2ffb53a1a642353ae108ffb496eba73c5f0/src/main/CoverageTrack.js#L38)):</sup>
This looks much better. Thanks for the change!

---

<sup>**[src/main/PileupTrack.js, line 322 \[r1\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/247#-JzC09ghRDvmy-8bg7gx)** ([raw file](https://github.com/hammerlab/pileup.js/blob/5978d2ffb53a1a642353ae108ffb496eba73c5f0/src/main/PileupTrack.js#L322)):</sup>
whoa! Did flow catch this? 

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/pileup.js/247)
<!-- Sent from Reviewable.io -->

danvk commented:
Review status: 6 of 8 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

<sup>**[lib/q.js, line 265 \[r1\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/247#-JzC0KHj-L-XAZX6P-yn)** ([raw file](https://github.com/hammerlab/pileup.js/blob/5978d2ffb53a1a642353ae108ffb496eba73c5f0/lib/q.js#L265)):</sup>
Sure. Added a comment w/ a link to the relevant issue.

---

<sup>**[src/main/CoverageTrack.js, line 27 \[r1\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/247#-JzC-xuPdGWXjib21L_h)** ([raw file](https://github.com/hammerlab/pileup.js/blob/5978d2ffb53a1a642353ae108ffb496eba73c5f0/src/main/CoverageTrack.js#L27)):</sup>
I think this is the remains of me bisecting a Flow error. I reverted the change.

---

<sup>**[src/main/PileupTrack.js, line 322 \[r1\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/247#-JzC09ghRDvmy-8bg7gx)** ([raw file](https://github.com/hammerlab/pileup.js/blob/5978d2ffb53a1a642353ae108ffb496eba73c5f0/src/main/PileupTrack.js#L322)):</sup>
It's not really a bug, just Flow not wanting to let you implicitly cast numbers to bools.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/pileup.js/247)
<!-- Sent from Reviewable.io -->

armish commented:
:shipit:

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/pileup.js/247)
<!-- Sent from Reviewable.io -->
